test/validators/isBefore.test.js

```diff
@@ -0,0 +1,90 @@
+import test from '../testFunctions';
+
+describe('isBefore', () => {
+  it('should validate dates against an end date', () => {
```

Comment:
I didn't do it for `isAfter` but maybe we can split this up in two `it` blocks? One without arguments (or at least that don't use the old nor new syntax like `[]`) and one with arguments. This way it's easier to see what the changes are of the two syntaxes and therefore the scenarios that we need to replicate
Just a new idea, so I'm open for feedback and other ideas

Reply:
Sorry, I am not sure what you mean. Could you give an example with the new syntax?

Reply:
The new syntax is using the `comparisonDate` in the object.

So we would have;
```
describe('isBefore'
   it('should validate dates against the current date'
     test({
       validator: 'isBefore',
       valid: ,
       invalid:
     })
     test({
       validator: 'isBefore',
       args: [],
       valid: ,
       invalid:
      })
   it('should validate dates against an end date'
     test({
       validator: 'isBefore',
       args: [{ comparisonDate: ],
       valid: ,
       invalid:
      })
   describe('(legacy syntax)'
     it('should validate dates against an end date'
       test({
         validator: 'isBefore',
         args: ['08/04/2011'],
         valid: ,
         invalid:
        })
```

Is that clear?

Reply:
Well, when you look at the three new tests, those will also "should validate dates against the current date", with two of those w/ arguments. So the test description will not be consistent. 

I've just pushed a proposed refactor to the tests. Let me know what you think :). Here you can also see how important it is to write test cases for all possible scenarios and permutations, ideally upfront, before implementation. That would have made it easier to refactor the implementation to the new syntax. The existing tests probably still lack possible values or permutations. We should probably also test against other nullish values. E.g. should we guard against `null`? Because the code will break if `options === null`.